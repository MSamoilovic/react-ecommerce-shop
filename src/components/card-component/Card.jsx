import React from "react";

const Card = (props) => (
  <div className="Random Div">
    <h1>{props.name}</h1>
  </div>
);

export default Card;
