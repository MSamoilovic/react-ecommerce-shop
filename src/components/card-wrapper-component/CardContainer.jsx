import React from "react";

import Card from "../card-component/Card";

const CardContainer = () => (
    <div>
      <Card name={'Marko'} />
    </div>
  );

export default CardContainer;
