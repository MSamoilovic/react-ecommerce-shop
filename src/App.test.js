import React from "react";
//import { render } from '@testing-library/react';
import { shallow, mount } from "enzyme";
import Card from "./components/card-component/Card";
import CardContainer from "./components/card-wrapper-component/CardContainer";

const mockName = "marko";
const cardWrapper = shallow(<Card name={mockName} />);
const cardContainerWrapper = mount(<CardContainer />);


describe("Card Component", () => {
  it("should render", () => {
    expect(cardWrapper).toMatchSnapshot();
  });

  it("should display props correctly", () => {
    expect(cardWrapper.find("h1").html()).toEqual(`<h1>${mockName}</h1>`);
    expect(cardWrapper.text()).toEqual(mockName);
    expect(cardWrapper.find("h1").text()).toEqual(mockName);
  });
});

describe("CardContainer component", () => {
  it("should render", () => {
    expect(cardContainerWrapper).toMatchSnapshot();
  });
});
