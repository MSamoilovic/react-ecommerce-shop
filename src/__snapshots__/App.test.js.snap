// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render 1`] = `ShallowWrapper {}`;

exports[`CardContainer component should render 1`] = `ReactWrapper {}`;
